tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordEntry.cs

```diff
@@ -92,29 +91,50 @@ private static byte[] DeserializeBody(IDictionary<string, string[]> headers, in
 
             if (IsTextContentType(headers, out Encoding encoding))
             {
-                if (property.ValueKind == JsonValueKind.Object)
-                {
-                    using var memoryStream = new MemoryStream();
-                    using var writer = new Utf8JsonWriter(memoryStream, writerOptions);
-                    property.WriteTo(writer);
-                    writer.Flush();
-                    return memoryStream.ToArray();
-                }
-                else if (property.ValueKind == JsonValueKind.Array)
-                {
-                    StringBuilder stringBuilder = new StringBuilder();
+                using var memoryStream = new MemoryStream();
+                using var writer = new Utf8JsonWriter(memoryStream, WriterOptions);
 
-                    foreach (JsonElement item in property.EnumerateArray())
+                switch (property.ValueKind)
+                {
+                    case JsonValueKind.Object:
+                        property.WriteTo(writer);
```

Comment:
I'd rewrite this method so it extracts byte[] first and then tried to reformat it if it's JSON.

RIght now all the fallbacks are hard to follow.